Project/Assets/_Scripts/_new_Scripts/Test/AnimationConfigurator.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AnimationConfigurator : MonoBehaviour
+{
+    Animator animator;
+    Camera cam;
+    
+    float y;
+    float targetY;
+    float remainingY;
+    bool isFacingRight;
+    float rotationSpeed;
+    float deltaTime;
+    Vector3 headDir;
+    Vector3 headPosition;
+    float headAngle;
+
+    void Start()
+    {
+        animator = GetComponentInChildren<Animator>();
+        cam = GameObject.Find("Camera").GetComponent<Camera>();
+        y = 0.0f;
+        rotationSpeed = 700.0f;
+    }
+
+
+    void Update()
+    {
+        deltaTime = Time.deltaTime;
+        isFacingRight = animator.GetBool("isFacingRight");
+        headPosition = cam.WorldToScreenPoint(animator.GetBoneTransform(HumanBodyBones.Head).position);
+        headDir = cam.WorldToScreenPoint(headPosition - Input.mousePosition);
+        headAngle = Vector3.Angle(headDir, transform.up * -1.0f);
+
+        if (Input.mousePosition.x - 20.0f >= headPosition.x)
+        {
+            animator.SetFloat("headTiltPercent", (headAngle - 20.0f) / 180.0f);
+        }
+
+        if (Input.mousePosition.x - 50.0f >= headPosition.x && isFacingRight == false)
+        {
+            animator.SetBool("isFacingRight", true);  
+        }
+
+        if (Input.mousePosition.x + 20.0f <= headPosition.x)
+        {
+            animator.SetFloat("headTiltPercent", (200.0f - headAngle) / 180.0f);

```

Comment:
Te wywołania tych funkcji też można by opakować w funkcje.
180 stopni można by jako prywatną stałą wstawić

Reply:
Mogę to jak najbardziej zrobić, tylko jest pytanie, czy na tym nam zależy. Gdy opakuję tą metodę w inną metodę, to uzyskujemy inną równoważną metodę, a metoda SetFloat() dalej będzie publiczna. Zyskamy na przejrzystości kodu, ale mając niejednoznaczność poprzez posiadanie dwóch metod, które robią dokładnie to samo, tylko jedna jest zwykłym opakowaniem drugiej także dostępnej, łamiemy hermetyzację. Taki mankament Unity i generalnie pisania skryptów zamiast solidnego obiektowego szkieletu od góry do dołu.

Reply:
Ale jeśli ją nazwiesz dobrze to ja będę wiedzieć co ona robi a tak to nie wiem 😞

Reply:
No chyba, że tak.